### PR TITLE
T1598

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3fb0271cda5024f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3fb0271cda5024f/output.txt

Total script time: 0.81 mins

File: src/core/media.js [hunk 1]
```diff
@@ -465,6 +482,7 @@
       function retrieveSrc() {
         var targetElement = document.getElementById( _target ),
             url = "";
+            console.log( _target, targetElement );
```
Comment (ScottDowne) [inline src/core/media.js]: Remove this.

File: src/core/views/media-view.js [hunk 2]
```diff
@@ -180,27 +180,37 @@ define( [ "ui/page-element", "ui/logo-spinner", "util/lang", "ui/widget/textbox"
...
+      var url = media.url;
+      if( typeof( url ) === "string" ) {
+        if ( url.indexOf( "," ) > -1 ) {
+          url = url.split( "," );
+          parseURLArray( url );
         }
-        for ( var i = 0; i < urls.length; ++i ) {
-          currentUrls[ i ].value = urls[ i ];
+        else {
+          _urlTextbox.value = url;
```
Comment (ScottDowne) [inline src/core/views/media-view.js]: trailing whitespace.

File: src/core/media.js [hunk 3]
```diff
@@ -78,6 +78,23 @@
...
+              if (  targetElement && 
+                    targetElement.getAttribute( "data-butter-media-controls" ) ) {
+                // Check if the media element is the target itself, or if it's a child node
+                if (  [ "VIDEO", "AUDIO" ].indexOf( targetElement.nodeName ) === -1 && 
+                      targetElement.firstChild &&
+                      [ "VIDEO", "AUDIO" ].indexOf( targetElement.firstChild.nodeName ) > -1 ) {
+                  targetElement.firstChild.setAttribute( "controls" );
+                }
+                else {
+                  targetElement.setAttribute( "controls" );
```
Comment (ScottDowne) [inline src/core/media.js]: Trailing whitespace.

File: src/core/media.js [hunk 4]
```diff
@@ -78,6 +78,23 @@
...
                 _view.update();
               }
+
+              // If the target element has a `data-butter-media-controls` property,
+              // set the `controls` attribute on the corresponding media element.
+              var targetElement = document.getElementById( _target );
+              if (  targetElement && 
+                    targetElement.getAttribute( "data-butter-media-controls" ) ) {
+                // Check if the media element is the target itself, or if it's a child node
+                if (  [ "VIDEO", "AUDIO" ].indexOf( targetElement.nodeName ) === -1 &&
```
Comment (ScottDowne) [inline src/core/media.js]: Trailing whitespace.

File: src/core/media.js [hunk 5]
```diff
@@ -78,6 +78,23 @@
               if( _view ){
                 _view.update();
               }
+
+              // If the target element has a `data-butter-media-controls` property,
+              // set the `controls` attribute on the corresponding media element.
+              var targetElement = document.getElementById( _target );
+              if (  targetElement &&
```
Comment (ScottDowne) [inline src/core/media.js]: Trailing whitespace.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/eb8086ead868ec1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/eb8086ead868ec1/output.txt

Total script time: 0.91 mins

Server is running on http://pr553.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/0f1ea77fedec8cb/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/2979bd0a9f3fe95/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/0f1ea77fedec8cb/output.txt

Total script time: 0.79 mins

Server is running on http://pr553.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2979bd0a9f3fe95/output.txt

Total script time: 0.56 mins

File: src/core/views/media-view.js [hunk 6]
```diff
@@ -165,7 +165,7 @@ define( [ "ui/page-element", "ui/logo-spinner", "util/lang", "ui/widget/textbox"
 
     }
 
-    function testUrl(url) {
+    function testUrl ( url ) {
```
Comment (mjschranz) [inline src/core/views/media-view.js]: I don't think we actually want that space padding before the arguments brackets.

File: src/core/views/media-view.js [hunk 7]
```diff
@@ -180,27 +180,31 @@ define( [ "ui/page-element", "ui/logo-spinner", "util/lang", "ui/widget/textbox"
     _logoSpinner.start();
     _changeButton.setAttribute( "disabled", true );
 
-    function updateURLS(){
-      if( typeof( media.url ) === "string" ) {
-        _urlTextbox.value = media.url;
+    function parseURLArray ( urlArray ) {
```
Comment (mjschranz) [inline src/core/views/media-view.js]: same as above

File: src/core/views/media-view.js [hunk 8]
```diff
@@ -180,27 +180,31 @@ define( [ "ui/page-element", "ui/logo-spinner", "util/lang", "ui/widget/textbox"
...
+      while ( currentUrls.length > urlArray.length ) {
+        removeUrl( currentUrls[ currentUrls.length - 1 ] );
+        currentUrls = _urlList.querySelectorAll( "input[type='text']" );
+      }
+      for ( var i = 0; i < urlArray.length; ++i ) {
+        currentUrls[ i ].value = urlArray[ i ];
+      }
+    }
+
+    function updateURLS () {
```
Comment (mjschranz) [inline src/core/views/media-view.js]: same as above

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/01c7630c6855c5e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/75eb76dba1b9482/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/01c7630c6855c5e/output.txt

Total script time: 1.33 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/75eb76dba1b9482/output.txt

Total script time: 1.33 mins

**Pull request merged**